### DISCUSSION TITLE
Rendering structured data in templates

### DISCUSSION THREAD

**Discussion by madduck** [requarks/wiki#7465] [Help / Questions] [answered]
One of the features of [Ikiwiki](https://ikiwiki.info/) is that I can create a page with data, e.g.

```
Color: blue
Band: The Beatles
Day: Wednesday
```

and render that page with a custom template, e.g.:

```
Today is {{Day}}, I am feeling {{Color}}, and listening to {{Band}}.
```

If I have many data pages like this, then it's trivial to make a change to the look of all of them by adapting the template.

Is this a feature worth entertaining? How hard could it be to implement this?

Thanks,
m

**Comment by NGPixel** [comment, marked as answer]
Follow https://feedback.js.wiki/wiki/p/templates-content-transclusion-infoboxes

**Reply by madduck** [reply]
Thanks, althought I am not sure transclusion encompasses the feature I mean. Transclusion means including a file, right? I want to be able to do more. In the above example, there is no file to include that defines `{{Day}}` and another that defines `{{Color}}`. Both of those are defined in a data page, and there will be other data pages. When a data page is loaded, its fields are made available to the template, and the template is rendered in the context of these fields of the data pages. Does this make sense?

**Reply by NGPixel** [reply]
Yes, this is part of that post. Both template inclusion and specific data fields will be supported.

**Comment by madduck** [comment]
Fantastic, thank you!
